readiness-tool/popup.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * @fileoverview Entry point for ART Chrome extension. 
+ * Scans the current page against the vendors in the app.json file and
+ * specifies them as either supported or not supported.
+ */
+
+self.popups = {};
+self.popups.isSupported = isSupported;
+self.popups.doesRegexExist = doesRegexExist;
+
+let supportedAds;
+let supportedAnalytics;
+let notSupportedAds;
+let notSupportedAnalytics;
+let notSupportedOther;
+const loadingMessage = 'Loading...';
+const blankMessage = '';
+
+/**
+ * Callback function that sends a response upon receiving message
+ * @param {!Object} request - Message Object
+ * @param {!Object} sender - Message sender defined here
+ */
+chrome.runtime.onMessage.addListener(function(request, sender) {
+  if (request.action == 'getSource') {
+    // htmlOfTab contains the page source as a string
+    htmlOfTab = request.source;
+    findDetectedApps(htmlOfTab);
+  }
+});
+
+window.onload = function onWindowLoad() {
+  // When page is loaded, display 'Loading...' so the user expects content
+  supportedAds = document.getElementById('ads-supported');
+  supportedAnalytics = document.getElementById('analytics-supported');
+  notSupportedAds = document.getElementById('ads-notSupported');
+  notSupportedAnalytics = document.getElementById(
+    'analytics-notSupported');
+  notSupportedOther = document.getElementById('other-notSupported');
+  supportedAds.innerHTML = supportedAnalytics.innerHTML =
```

Comment:
Recommend setting `Node.textContent` instead. In general, innerHTML has the risk of script injection.